### ISSUE TITLE
SPIKE: Alternatives to ModSecurity

### ISSUE THREAD

**Issue by AntonyBishop** [ministryofjustice/cloud-platform#2340]
We investigated alternatives for WAF as part of #1026. We opted for ModSecurity to provide this feature.

We are now hitting issues of scaling with ModSecurity. 

We want to investigate what other alternatives are available and whether it is worth moving to another product?

**Comment by vijay-veeranki** [comment]
https://www.g2.com/products/modsecurity/competitors/alternatives

Comments from Ross AWS: 

AWS WAF and ALB work at L7 while NLB operates at L4. I have suggested looking at the ALB ingress controller to then use WAF, particularly as it now supports multiple ingresses per ALB. You could put CloudFront in front of your NLB and enable WAF for CF, which would apply firewall at the edge and give you other advantages of caching content but will also increase complexity and the number of layers.

**Comment by vijay-veeranki** [comment]
When we initially looked the options for WAF, given the nature of our setup (NLB -> nginx-ingress -> application), we're somewhat limited in our options. Considering the below options we opted for mod-sec speaking to security team

https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#lua-resty-waf
https://kubernetes.github.io/ingress-nginx/user-guide/third-party-addons/modsecurity/
https://wallarm.com/solutions/waf-for-kubernetes


We had some issues with mod-sec, looked at other alternatives as part of this ticket.

Other tools need to be maintained separately and now we know the issue with mod-sec and that comes as part of ingress-controller we don't need to maintain another component for WAF.

Closing this ticket and we need to re-visit it if needed.
